redis.go

```diff
@@ -0,0 +1,72 @@
+package iprepd
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/go-redis/redis"
+	log "github.com/sirupsen/logrus"
+)
+
+type redisLink struct {
+	master      *redis.Client
+	readClients []*redis.Client
+}
+
+func (r *redisLink) del(k ...string) *redis.IntCmd {
+	return r.master.Del(k...)
+}
+
+func (r *redisLink) flushAll() *redis.StatusCmd {
+	return r.master.FlushAll()
+}
+
+func (r *redisLink) get(k string) (ret []byte, err error) {
+	readRot := make([]*redis.Client, len(r.readClients))
+	p := rand.Perm(len(r.readClients))
+	for a, b := range p {
+		readRot[b] = r.readClients[a]
+	}
+	for i := range readRot {
```

Comment:
Unless I'm mistaking your intentions here you can range over `p` instead of creating an array and populating it:

```
p := rand.Perm(len(r.readClients))
for _, i := range p {
  ret, err = readRot[i].Get(k).Bytes()
  ...
}
```

Reply:
👍